public/js/src/api.js

```diff
@@ -0,0 +1,53 @@
+const POLITICS_URL = `https://content.guardianapis.com/search?from-date=${yesterday()}&to-date=${today()}&q=politics&api-key=test&show-fields=all`
+
+const frontPage = new FrontPage();
+
+fetchArticles();
+
+function fetchArticles(url = POLITICS_URL) {
+  if (ENVIRONMENT === 'test') {
+    const results = '{"response":{"status":"ok","results":"Test OK"}}'
+    frontPage.articles = results;
+    return results;
+  }
+  fetch(url, {
+    method: 'GET',
+    mode: 'cors',
+    headers: {
+      'Content-Type': 'application/json'
+    }
+  }).then(response => response.json())
+    .then(data => frontPage.articles = data.response.results);
+}
+
+function today() {
+  const today = new Date().toLocaleDateString();
+  return today.split('/').reverse().join('-')
+}
+
+function yesterday() {
+  const yesterday = new Date();
+  yesterday.setDate(yesterday.getDate() - 1)
+  return yesterday.toLocaleDateString().split('/').reverse().join('-')
+}
+
+function fetchSummaries(articleUrl) {
+  const makersUrl = 'http://news-summary-api.herokuapp.com/aylien?apiRequestUrl='
+  const aylienUrl = 'https://api.aylien.com/api/v1/summarize?url='
+  const url = makersUrl + aylienUrl + articleUrl
+
+  if (ENVIRONMENT === 'test') {
+    const result = '{"response":{"status":"ok","results":"Test OK"}}'
+    frontPage.summaries.push(result);
```

Comment:
For better encapsulation, code that's outside of the `FrontPage` class shouldn't interact directly with frontPage.summaries. This makes it more onerous to change the internals of `FrontPage` later if you chose to stores the summaries in a different data structure.

And actually ... directly referring to the `FrontPage` class from this file at all makes your API layer code and your model layer code quite intertwined. I think it would be more cleanly separated if you had something like a controller (like you sort have inside `interface.js`) that calls `fetchArticles` and `fetchSummaries` and passes the resulting data into a `FrontPage` class.

Reply:
cool thanks, I think I see what you mean but may have to ask you further about this one :) 

Do you mean that `api.js` passes it's data to a separate file / class, say `controller.js`, which then adds it to `frontpage.js`?

Reply:
Sort of, I mean that in `controller.js` (or in your current `interface.js`) you could have something like:

```
let frontpage = new FrontPage();

fetchArticles(data => frontpage.addArticles(data))
 
# ... code to transform the contents of frontpage into HTML ...
```

where `fetchArticles` is a function inside `api.js`

Reply:
ah ok, got it, thanks!